### PR TITLE
Adding http test for system header

### PR DESCRIPTION
**Pull request by vkarak1**

### CODE REVIEW

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+            wallet=self.wallet,
+            shell=self.shell,
+            endpoint=self.cluster.default_rpc_endpoint,
+            rule=self.PLACEMENT_RULE,
+            basic_acl=PUBLIC_ACL,
+        )
+
+    @pytest.fixture(scope="class")
+    @allure.title("return_epoch_duration in seconds")
+    def return_epoch_duration(self) -> int:
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:60]: naming is incorrect.
`return_epoch_duration` -> `epoch_duration`

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    PLACEMENT_RULE = "REP 2 IN X CBF 1 SELECT 2 FROM * AS X"
+
+    @pytest.fixture(scope="class", autouse=True)
+    @allure.title("[Class/Autouse]: Prepare wallet and deposit")
+    def prepare_wallet(self, default_wallet):
+        Test_http_system_header.wallet = default_wallet
+
+    @pytest.fixture(scope="class")
+    @allure.title("Create container")
+    def create_container(self):
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:49]: `create_container` -> `user_container` or something

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+            endpoint=self.cluster.default_rpc_endpoint,
+            shell=self.shell,
+        )
+        return net_info["epoch_duration"]
+
+    @allure.title("Return N-epoch count in minutes")
+    def epoch_count_into_mins(self, return_epoch_duration, epoch: int) -> str:
+        mins: int
+        mins = return_epoch_duration * epoch / 60
+        return f"{(mins)}m"
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:72]: why `mins` needs to be covered into warm `( )` ?  :)

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    def return_epoch_duration(self) -> int:
+        net_info = get_netmap_netinfo(
+            wallet=self.wallet,
+            endpoint=self.cluster.default_rpc_endpoint,
+            shell=self.shell,
+        )
+        return net_info["epoch_duration"]
+
+    @allure.title("Return N-epoch count in minutes")
+    def epoch_count_into_mins(self, return_epoch_duration, epoch: int) -> str:
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:69]: `epoch_duration` is probably `epoch_duration: int`

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+        mins: int
+        mins = return_epoch_duration * epoch / 60
+        return f"{(mins)}m"
+
+    @allure.title("Return future timestamp after N epochs are passed")
+    def epoch_count_into_timestamp(
+        self, return_epoch_duration, epoch: int, rfc3339: Optional[bool] = False
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:76]: epoch_duration is probably epoch_duration: int

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+        net_info = get_netmap_netinfo(
+            wallet=self.wallet,
+            endpoint=self.cluster.default_rpc_endpoint,
+            shell=self.shell,
+        )
+        return net_info["epoch_duration"]
+
+    @allure.title("Return N-epoch count in minutes")
+    def epoch_count_into_mins(self, return_epoch_duration, epoch: int) -> str:
+        mins: int
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:70]: redundant

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+        current_datetime = datetime.datetime.utcnow()
+        epoch_count_in_seconds = return_epoch_duration * epoch
+        future_datetime = current_datetime + datetime.timedelta(seconds=epoch_count_in_seconds)
+        if rfc3339:
+            return future_datetime.isoformat("T") + "Z"
+        else:
+            return str(calendar.timegm(future_datetime.timetuple()))
+
+    @allure.title("Return current epoch#")
+    def return_curr_epoch(self) -> int:
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:87]: why this "proxy" is required?)

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+        self, create_container: str, object_size: int, return_epoch_duration: int
+    ):
+        self.tick_epoch()
+        epoch_count = 2
+        expected_epoch = self.return_curr_epoch() + epoch_count
+        logger.info(
+            f"epoch duration={return_epoch_duration}, current_epoch= {self.return_curr_epoch()} expected_epoch {expected_epoch}"
+        )
+        keys = [NEOFS_EXPIRATION_TIMESTAMP, NEOFS_EXIPRATION_RFC3339]
+        values = [
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:341]: why not just 
```
attributes = {
     ... : ...
}
```

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+    def test_http_rfc_object_unavailable_after_expir(
+        self, create_container: str, object_size: int, return_epoch_duration: int
+    ):
+        self.tick_epoch()
+        epoch_count = 2
+        expected_epoch = self.return_curr_epoch() + epoch_count
+        logger.info(
+            f"epoch duration={return_epoch_duration}, current_epoch= {self.return_curr_epoch()} expected_epoch {expected_epoch}"
+        )
+        keys = [NEOFS_EXIPRATION_RFC3339]
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:385]: same, 
why not just 
```
attributes = {
     ... : ...
}
```

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+    @pytest.mark.parametrize(
+        "object_size",
+        [pytest.lazy_fixture("simple_object_size")],
+        ids=["simple object"],
+    )
+    def test_unable_put_expired_rfc(self, create_container: str, object_size: int):
+        headers = attr_into_str_header_curl({"Neofs-Expiration-RFC3339": "2021-11-22T09:55:49Z"})
+        file_path = generate_file(object_size)
+        with allure.step(
+            "[negative] Put object using HTTP with attribute Neofs-Expiration-RFC3339 where duration is in the past"
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:228]: redundant title here. We alread have it as title of the test

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+                headers=headers,
+                error_pattern=f"{EXPIRATION_TIMESTAMP_HEADER} must be in the future",
+            )
+
+    @allure.title(
+        "[negative] Put object using HTTP with attribute Neofs-Expiration-RFC3339 where duration is in the past"
+    )
+    @pytest.mark.parametrize(
+        "object_size",
+        [pytest.lazy_fixture("simple_object_size")],
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:221]: if this is the only one parameter, why not just use it as arg of the test

```
@pytest.mark.parametrize(
        "object_size",
        [pytest.lazy_fixture("simple_object_size")],
        ids=["simple object"],
    )
    def test_unable_put_expired_rfc(self, create_container: str, object_size: int):
```
just 

```
def test_unable_put_expired_rfc(self, create_container: str, simple_object_size: int):
```

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+        )
+        return oid, head
+
+    @allure.title("[negative] attempt to put object with expired epoch")
+    @pytest.mark.parametrize(
+        "object_size",
+        [pytest.lazy_fixture("simple_object_size")],
+        ids=["simple object"],
+    )
+    def test_unable_put_expired_epoch(self, create_container: str, object_size: int):
```
Comment (abereziny) [inline pytest_tests/testsuites/services/http_gate/test_http_system_header.py:158]: we can use directly simple_object_size as argument in the test if it's only one param

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 1 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:60]: done

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 3 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:72]: good catch, thanks!

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 4 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:69]: corrected, thanks!

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 6 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:70]: corrected thanks!

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 5 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:76]: corrected!:) thanks!

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 8 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:341]: corrected)

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 9 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:385]: corrected

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 10 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:228]: done)

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 11 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:221]: done

File: pytest_tests/testsuites/services/http_gate/test_http_system_header.py (hunk 12 above)
Comment (vkarak1) [reply pytest_tests/testsuites/services/http_gate/test_http_system_header.py:158]: done

**Review by abereziny — APPROVED** [review]

**Review by anikeev-yadro — APPROVED** [review]

**Review by dansingjulia — APPROVED** [review]

**Pull request merged**